### ISSUE TITLE
guv-web Error: Restart / Stop Not Working

### ISSUE THREAD

**Issue by dinosaw** [tableflip/guvnor#48]
When attempting to restart and/or stop processes via guv-web, I receive a vague error stating:
```
Stop error

process on HOST has failed to stop - undefined
```
or
```
Restart error

process on HOST failed to restart - undefined
```

At first I thought it was a permissions issue since I am running the processes as root:root, however I verified that I am also running guv-web as root - so permissions should not be a factor.

**Comment by achingbrain** [comment]
This usually happens when an `Error` is emitted that has no `message` property.

There are logs under /var/log/guvnor - could you please have a look at guvnor.log, guvnor.error.log and the log for your process and post anything relevant?

**Comment by dinosaw** [comment]
I had assumed Guvnor would generate some error logs so I dove into `/var/log/guvnor` earlier and did not see any.

Now with you confirming there should be an error log, I looked again and I do not see any. Also despite today being the `2015-03-16` and me having logged into the web interface, the last log entry for `guvnor-web.log` is `2015-03-15` (I'm assuming this is the web interface log as this is what I ran *ouroboros* style via `guv web`).

I'm not sure what exactly is going wrong here unless error logging is not enabled by default and I have to enable/define myself?

Is there any more information I can provide?

**Comment by dinosaw** [comment]
Also perhaps I should mention this. I noticed that some log files are owned with a different group than others, as well as the main log directory is owned by the guvnor group. I went ahead and removed all log files that were present, killed guvnor, and started it back up again. This is basically the structure of `/var/log/guvnor` now:

```
drwxrwx---   2 root guvnor 4.0K Mar 16 13:08 .
drwxr-xr-x. 13 root root   4.0K Mar 15 03:17 ..
-rw-------   1 root root   4.2K Mar 16 13:09 ghost-blog-1.log.2015-03-16
-rw-------   1 root root    392 Mar 16 13:09 ghost-blog-2.log.2015-03-16
-rw-------   1 root guvnor  140 Mar 16 13:00 guvnor.log.2015-03-15
-rw-------   1 root guvnor 4.5K Mar 16 13:09 guvnor.log.2015-03-16
-rw-r--r--   1 root root   8.4K Mar 16 13:09 guvnor-web.log.2015-03-16
```

The changes in user/group ownership seems inconsistent to me. So I'm wondering if this could be causing my problem? Here is a `ps aux | grep guv`. Not sure if that will provide any insight, but here.

```
root     11920  0.7  6.5 793944 66920 ?        Ssl  13:08   0:04 guvnor                                                        
root     11922  0.1  2.6 723716 26864 ?        Sl   13:08   0:00 guvnor - node-inspector                                        
root     11928  6.0 21.9 1223356 223504 ?      Sl   13:08   0:33 guvnor-web                                                                             
root     12001  0.0  0.0 103316   832 pts/3    S+   13:18   0:00 grep guv
```

**Comment by achingbrain** [comment]
That looks ok to me - the daemon runs as root:guvnor, hence the owner of guvnor.log, but you've probably started guv-web as root, who's default group is root, hence guvnor-web.log is owned by root:root.  Incidentally you don't need to run guvnor-web as root, any user will do.

I've just release a new version that should resolve the undefined error message - could you please give it a go?

**Comment by dinosaw** [comment]
Hi **achingbrain**, thanks for the follow up.

I gotta say, after you adding in the error message, I was able to figure out what I was doing wrong. It turns out it was user error (big surprise, right?). It's now working as it should.

I had overlooked a piece of documentation where you assign a user to have privileges for a certain host (i.e. adding the host / secret to `guvnor-web-users`.

I appreciate your time and you helping develop such a great piece of software!
